ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/SavePreMergeHeadersStep.java

```diff
@@ -32,15 +34,39 @@
 public class SavePreMergeHeadersStep implements Function<BlockHeader, Stream<BlockHeader>> {
   private static final Logger LOG = LoggerFactory.getLogger(SavePreMergeHeadersStep.class);
   private final MutableBlockchain blockchain;
-  private final long mergeBlockNumber;
+  private final boolean isPoS;
+  private final long lastPoWBlockNumber;
+  private final long checkpointBlockNumber;
+  private final Optional<ConsensusContext> consensusContext;
 
   private final AtomicBoolean shouldLog = new AtomicBoolean(true);
   private static final int LOG_REPEAT_DELAY_SECONDS = 30;
   private static final int LOG_PROGRESS_INTERVAL = 1000;
 
-  public SavePreMergeHeadersStep(final MutableBlockchain blockchain, final long mergeBlockNumber) {
+  public SavePreMergeHeadersStep(
+      final MutableBlockchain blockchain,
+      final boolean isPoS,
+      final long checkpointBlockNumber,
+      final Optional<ConsensusContext> consensusContext) {
     this.blockchain = blockchain;
-    this.mergeBlockNumber = mergeBlockNumber;
+    this.isPoS = isPoS;
+    this.checkpointBlockNumber = checkpointBlockNumber;
+    this.lastPoWBlockNumber = checkpointBlockNumber - 1;
+    this.consensusContext = consensusContext;
+  }
+
+  public static SavePreMergeHeadersStep createForPoS(
+      final MutableBlockchain blockchain,
+      final long firstPoSBlockNumber,
+      final Optional<ConsensusContext> consensusContext) {
+    return new SavePreMergeHeadersStep(blockchain, true, firstPoSBlockNumber, consensusContext);
+  }
+
+  public static SavePreMergeHeadersStep createForPoA(
+      final MutableBlockchain blockchain,
+      final long checkpointBlockNumber,
+      final Optional<ConsensusContext> consensusContext) {
+    return new SavePreMergeHeadersStep(blockchain, false, checkpointBlockNumber, consensusContext);
```

Comment:
nit: remove the static creators and simplify the pipeline since we are passing isPoS